Classes/Configuration/ExtensionBuilderConfigurationManager.php

```diff
@@ -201,58 +202,23 @@ public static function getExtensionBuilderJson($extensionKey, $storagePath = nul
-     *
-     * @return array with configuration values
-     * @throws \TYPO3\CMS\Extbase\Configuration\Exception\InvalidConfigurationTypeException
+     * @param $className string
+     * @return string
      */
-    public function getExtbaseClassConfiguration($className): array
+    public function getPersistenceTable($className)
     {
-        $classConfiguration = [];
-        if (strpos($className, '\\') === 0) {
-            $className = substr($className, 1);
-        }
-        $frameworkConfiguration = $this->configurationManager->getConfiguration($this->configurationManager::CONFIGURATION_TYPE_FRAMEWORK);
-        $classSettings = $frameworkConfiguration['persistence']['classes'][$className];
-        if ($classSettings !== null) {
-            if (isset($classSettings['subclasses']) && is_array($classSettings['subclasses'])) {
-                $classConfiguration['subclasses'] = $classSettings['subclasses'];
-            }
-            if (isset($classSettings['mapping']['recordType']) && strlen($classSettings['mapping']['recordType']) > 0) {
-                $classConfiguration['recordType'] = $classSettings['mapping']['recordType'];
-            }
-            if (isset($classSettings['mapping']['tableName']) && strlen($classSettings['mapping']['tableName']) > 0) {
-                $classConfiguration['tableName'] = $classSettings['mapping']['tableName'];
-            }
-            $classHierachy = array_merge([$className], class_parents($className));
-            $columnMapping = [];
-            foreach ($classHierachy as $currentClassName) {
-                if (in_array(
-                    $currentClassName,
-                    [
-                        AbstractEntity::class,
-                        AbstractValueObject::class
-                    ]
-                )) {
-                    break;
-                }
-                $currentClassSettings = $frameworkConfiguration['persistence']['classes'][$currentClassName];
-                if ($currentClassSettings !== null) {
-                    if (isset($currentClassSettings['mapping']['columns']) && is_array($currentClassSettings['mapping']['columns'])) {
-                        ArrayUtility::mergeRecursiveWithOverrule(
-                            $columnMapping,
-                            $currentClassSettings['mapping']['columns'],
-                            0,
-                            // false means: do not include empty values form 2nd array
-                            false
-                        );
-                    }
-                }
-            }
-        }
-        return $classConfiguration;
+        $objectManager = GeneralUtility::makeInstance(ObjectManager::class);
+        return $objectManager->get(\TYPO3\CMS\Extbase\Persistence\Generic\Mapper\DataMapper::class)->getDataMap($className)->getTableName();
+    }
+
+    /**
+     * @param $className
+     * @return string
+     */
+    public function getRecordType($className)
```

Comment:
why not adding type hints here, would spare the phpdoc, which does not convey extra information then

Reply:
I agree - type hints would be less verbose
On the other hand all other methods use phpdoc in this class, so I would leave it like this until we refactor all methods.
More relevant: we should consider to change that for the generated classes and methods...

Reply:
Definitely. All generated code should use the maximum features of PHP 7.4 in terms of typehints. All redundant phpdoc can of course then be removed. 

`public function someThingUseful(string $goodName, MyClass $bestName): Magic {}`
Does not need a phpdoc unless the function needs extra explanation for the inner working or the parameters.

Reply:
So I suggest to merge this PR and change all the phpdoc related stuff in a separate commit, ok?

Reply:
Of course!